packages/engineer/feature/dev-server.dev-server.env.ts

```diff
@@ -0,0 +1,282 @@
+            }
+
+            const { port, app, close, socketServer } = await launchHttpServer({
+                staticDirPath: application.outputPath,
+                httpServerPort,
+            });
+            disposables.add(() => close());
+
+            const host = new WsServerHost(socketServer.of(`/${devServerEnv.env}`));
+
+            communication.clearEnvironment(devServerEnv.env);
+            communication.registerMessageHandler(host);
+            communication.registerEnv(devServerEnv.env, host);
+
+            const { features, configurations, packages } = application.analyzeFeatures();
+            if (singleFeature && featureName) {
+                application.filterByFeatureName(features, featureName);
+            }
+
+            const compiler = application.createCompiler({
+                mode,
+                features,
+                featureName,
+                configName,
+                publicPath,
+                title,
+                configurations,
+                staticBuild: false,
+                publicConfigsRoute,
+                overrideConfig,
+                singleFeature,
+            });
+
+            // This hack is to squeeze some more performance, because we can server the output in memory
+            // It was once a crash, which is no longer relevant
+            if (singleRun) {
+                for (const childCompiler of compiler.compilers) {
+                    childCompiler.watch = function watch(_watchOptions, handler) {
+                        childCompiler.run(handler);
+                        return {
+                            close(cb) {
+                                if (cb) {
+                                    cb();
+                                }
+                            },
+                            invalidate: () => undefined,
+                        };
+                    };
+                }
+            }
+
+            //Node environment manager, need to add self to the topology, I thing starting the server and the NEM should happen in the setup and not in the run
+            // So potential dependants can rely on them in the topology
+            const nodeEnvironmentManager = new NodeEnvironmentsManager(socketServer, {
+                configurations,
+                features,
+                defaultRuntimeOptions,
+                port,
+                inspect,
+                overrideConfig,
+            });
+            nodeEnvManager = nodeEnvironmentManager;
+
+            disposables.add(() => nodeEnvironmentManager?.closeAll());
+
+            if (engineConfig && engineConfig.serveStatic) {
+                for (const { route, directoryPath } of engineConfig.serveStatic) {
+                    app.use(route, express.static(directoryPath));
+                }
+            }
+
+            const topologyOverrides = (featureName: string): Record<string, string> | undefined =>
```

Comment:
Low priority, perhaps we need to somehow actually add engineer into the topology and not override it?